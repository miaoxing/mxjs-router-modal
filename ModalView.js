import React from 'react';
import {Modal} from 'react-bootstrap';
import {withRouter} from 'react-router';
import PropTypes from 'prop-types';

export default @withRouter
class ModalView extends React.Component {
  static defaultProps = {
    length: 1,
  };

  static propTypes = {
    history: PropTypes.object,
    length: PropTypes.number,
    children: PropTypes.node,
  };

  state = {
    show: true,
  };

  handleHide = () => {
    this.setState({show: false});
  };

  handleExited = () => {
    this.props.history.go(-this.props.length);
  };

  render() {
    return (
      <Modal
        show={this.state.show}
        onHide={this.handleHide}
        onExited={this.handleExited}
        className="modal-right"
      >
        <Modal.Body className="page-content">
          {this.props.children}
        </Modal.Body>
      </Modal>
    );
  }
}
