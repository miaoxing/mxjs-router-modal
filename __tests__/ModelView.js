import {render} from '@testing-library/react';
import ModalView from '../ModalView';
import {MemoryRouter} from 'react-router';

describe('ModalView', () => {
  test('basic', () => {
    const result = render(<MemoryRouter>
      <ModalView/>
    </MemoryRouter>);
    expect(result.container).toMatchSnapshot();
  });
});

