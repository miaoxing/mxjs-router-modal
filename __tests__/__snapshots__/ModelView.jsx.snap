// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalView basic 1`] = `
<div
  aria-hidden="true"
/>
`;
